src/main/steps/edge-case/subject-details/content.ts

```diff
@@ -47,16 +48,19 @@ export const form: FormContent = {
       ],
       parser: body => covertToDateObject('subjectDateOfBirth', body as Record<string, unknown>),
       validator: value => {
-        if (isDateInputNotFilled(value as CaseDate)) {
-          return 'required';
-        }
-        if (
-          isDateInputInvalid(value as CaseDate) ||
-          isFutureDate(value as CaseDate) ||
-          isObsoleteDate(value as CaseDate)
-        ) {
-          return 'invalid';
-        }
+        const date = value as CaseDate;
+
+        return isDateInputInvalid(date) && isDateInputNotFilled(date)
+          ? 'invalidAndIncomplete'
+          : isDateInputInvalid(date)
+          ? isDateInputInvalid(date)
+          : isDateInputNotFilled(date)
+          ? isDateInputNotFilled(date)
+          : isFutureDate(date)
+          ? isFutureDate(date)
+          : isObsoleteDate(date)
+          ? isObsoleteDate(date)
+          : undefined;
```

Comment:
I think chaining ternary operators together is not very readable.

Reply:
Fixed now